### ISSUE TITLE
SSL cert host name check against alternative names fails

### ISSUE THREAD

**Issue by multi-io** [kelproject/pykube#109]
(running on a temporary container in the cluster)

```
root@ubuntu:/# pip freeze
Babel==2.3.4
backports.shutil-get-terminal-size==1.0.0
decorator==4.0.6
enum34==1.1.6
httplib2==0.9.2
ipython==5.1.0
ipython-genutils==0.1.0
jsonpath-rw==1.4.0
jsonpath-rw-ext==0.1.9
oauth2client==4.0.0
oauthlib==2.0.1
pathlib2==2.2.0
pbr==1.10.0
pexpect==4.2.1
pickleshare==0.7.4
ply==3.7
prompt-toolkit==1.0.9
ptyprocess==0.5.1
pyasn1==0.1.9
pyasn1-modules==0.0.8
Pygments==2.1.3
pykube==0.14.0
pytz==2014.10
PyYAML==3.12
requests==2.12.4
requests-oauthlib==0.7.0
rsa==3.4.2
scandir==1.4
simplegeneric==0.8.1
six==1.10.0
traitlets==4.3.1
tzlocal==1.3
wcwidth==0.1.7
root@ubuntu:/#
root@ubuntu:/# cat test.py
import operator; import pykube

api = pykube.HTTPClient(pykube.KubeConfig.from_service_account())
pods = pykube.Pod.objects(api)
ready_pods = filter(operator.attrgetter("ready"), pods)

root@ubuntu:/# python test.py
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    ready_pods = filter(operator.attrgetter("ready"), pods)
  File "/usr/local/lib/python2.7/dist-packages/pykube/query.py", line 130, in __iter__
    return iter(self.query_cache["objects"])
  File "/usr/local/lib/python2.7/dist-packages/pykube/query.py", line 120, in query_cache
    cache["response"] = self.execute().json()
  File "/usr/local/lib/python2.7/dist-packages/pykube/query.py", line 104, in execute
    r = self.api.get(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pykube/http.py", line 127, in get
    return self.session.get(*args, **self.get_kwargs(**kwargs))
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 501, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 497, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: hostname '10.96.0.1' doesn't match either of 'kubernetes', 'kubernetes.default', 'kubernetes.default.svc', 'kubernetes.default.svc.cluster.local', '10.0.4.5', '185.56.135.79', '10.96.0.1'
root@ubuntu:/#
```

As you can see, the queried IP (10.96.0.1) is in the list. I also verified that that's the actual "Alternative Names" list of the server's certificate[1]. So I guess this is a bug in the client.

[1]
```
root@ubuntu:/# openssl s_client -showcerts -connect 10.96.0.1:443 </dev/null
CONNECTED(00000003)
depth=0 CN = kube-apiserver
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = kube-apiserver
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/CN=kube-apiserver
   i:/CN=kubernetes
-----BEGIN CERTIFICATE-----
MIIDXjCCAkagAwIBAgIIQ/jBvuJbfxowDQYJKoZIhvcNAQELBQAwFTETMBEGA1UE
AxMKa3ViZXJuZXRlczAeFw0xNzAxMDkwMDQ4MThaFw0xODAxMDkwMDQ4MTlaMBkx
FzAVBgNVBAMTDmt1YmUtYXBpc2VydmVyMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A
MIIBCgKCAQEAtBPuR5Hnh8Ne8QJINM0VhawvxW/K4W9yYkXHg1CceyiEhOLXQ23v
xIE5ZGRTY5Nx71rX0WmEsB2n+FYv7IZn9QDUhCixdkYlCSqIzTN/QzSACG6Xhkhe
TiQresNZ7FpcuSAnuF0ErRN7HYxbvdFo/N0oWFgpK2ULWstAMuiKqywqxVi0GqZx
YAigGCK5WB9JEADXX5dRyw/wAG3u/6ve2LLlHCO8HBoGrSf/EgCy+9TP+MtGqp28
jY68JrupBCr6kk2wWD9BWpy6gavf+vYOvbRgksm6JORL0KmlLXbQRo4W+EBRzy8w
T3Sfhsyk8CvUTGj0d9wU3Vk3DFH74XWrjwIDAQABo4GtMIGqMA4GA1UdDwEB/wQE
AwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIweQYDVR0RBHIwcIIK
a3ViZXJuZXRlc4ISa3ViZXJuZXRlcy5kZWZhdWx0ghZrdWJlcm5ldGVzLmRlZmF1
bHQuc3ZjgiRrdWJlcm5ldGVzLmRlZmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWyHBAoA
BAWHBLk4h0+HBApgAAEwDQYJKoZIhvcNAQELBQADggEBAHDF9QFGIs+cbOa9d1Fr
+MdjM9ZD/J9Nne1C+gpgxDxGTqKW7OyilOR3IXjrnIs+5Gfd73WSanG1o0qO+8uI
gUeavDtDuSAgtIVIzpq+QSEwmKpUpYEUPYwUt+wJ+YlKB1GjHXM0EC3ck0Qx8YZE
PbKWHkn0b9EXLWUIm5zQeGQDoL3TX3PgYyME3s8mfw/fwpnoXZtTyaQZm0UAO5jI
dv6OPulzA0cb/tCg0yjPcIFZYCjkzjvQyzxtsyoq3Tht3G8GQR2U30OedP9WNW4e
v+WWovC0AtEdwNp4DwtEoxocYN6wORSTN2BFw5/Krp7zC364caBd7QPh0M2Vm09q
Jp8=
-----END CERTIFICATE-----
[....]
```

**Comment by brosner** [comment]
This issue should have been resolved in requests 2.12+ (by bundling a urllib 1.18+). I am not in the position to debug, but here is the relavant pykube commit: 0f7b5bcc92c7b55a1b22c2bfa1198f9577d0b6a9

**Comment by multi-io** [comment]
Thanks for the super quick response :) :1st_place_medal:  Switched to Python3 now, it works there.

Closing the issue.
